File: ios/ImagesHandling/ImagesHandling/Glasses/ImagesBatchScreen.swift
//Copyright (c) 2020 Everysight LTD

import Foundation
import EvsKit
import NativeEvsKit

class ImagesBatchScreen:Screen {

    private let text = Text()
    private var images : [Image]? = nil

    override func onCreate() {
        super.onCreate()
        
        text
            .setResource(font: Font(stockFont: .small))
            .setCenter(x: getWidth()/2)
            .setY(y: getHeight()/3)

        add(uiElement: text)
        images = [Image]()
        let img1 = Image(id: "blast")
            .setResource(image: ImgSrc(nameWithExtension: "blast.jpg", slot: .s1))
        images?.append(img1)
        
        let img2 = Image(id: "heart")
            .setResource(image: ImgSrc(nameWithExtension: "heart.jpg", slot: .s2))
        images?.append(img2)

        let img3 = Image(id: "ship")
            .setResource(image: ImgSrc(nameWithExtension: "ship.jpg", slot: .s3))
        images?.append(img3)
        
        let img4 = Image(id: "watermelon")
            .setResource(image: ImgSrc(nameWithExtension: "watermelon.jpg", slot: .s4))
        images?.append(img4)
        
        let img5 = Image(id: "orange")
            .setResource(image: ImgSrc(nameWithExtension: "orange.jpg", slot: .s5))
        images?.append(img5)
        
        if let imgs = images {
            requestResourcesUpload(resources: imgs.getResources())
        }
        text.setVisibility(isVisible: true)
        text.setText(text: "Loading...")
    }

    
    override func onResourceUploadResult(resource: UIResource) {
        text.setText(text: "\(resource.getResourceName()) was loaded")
    }

    override func onResourcesUploadEnd() {
        super.onResourcesUploadEnd()
        text.setVisibility(isVisible: false)
        var i = 0
        if let imageArray = images {
            imageArray.forEach { img in
                img.setX(x: Float(i)*50.0).setY(y: getHeight()/2)
                add(uiElement: img)
                i+=1
            }
        }
    }

}

extension Array where Element == Image {
    func getResources() -> [ImgSrc] {
        var srcs = [ImgSrc]()
        for i in self {
            if let s = i.getResource() {
                srcs.append(s)
            }
        }
        return srcs
    }
}

File: android/ImagesHandling/src/main/java/com/everysight/sdk/samples/imageshandling/ImagesBatchScreen.kt
package com.everysight.sdk.samples.imageshandling

import UIKit.app.Screen
import UIKit.app.data.*
import UIKit.app.resources.Font
import UIKit.app.resources.ImgSrc
import UIKit.app.resources.UIResource
import UIKit.widgets.Image
import UIKit.widgets.Text
import UIKit.widgets.UIResourceElement

class ImagesBatchScreen:Screen() {

    private val text = Text()

    override fun onCreate() {
        super.onCreate()
        text.setResource(Font.StockFont.Small).setCenter(getWidth()/2).setY(getHeight()/3)
        add(text)

        //your can upload custom fonts as well via the Text UI element
        requestResourcesUpload(screenImages)

    }

    override fun onResourcesUploadStart() {
        super.onResourcesUploadStart()
        text.setVisibility(true)
        text.setText("Loading...")
    }
    override fun onResourceUploadResult(resource: UIResource) {
        text.setText("${resource.getResourceName()} was loaded")
    }

    override fun onResourcesUploadEnd() {
        super.onResourcesUploadEnd()
        text.setVisibility(false)
        screenImages.forEachIndexed{index,imgSrc->
            add(Image().setResource(imgSrc).setX(index*50f).setY(getHeight()/2))
        }

    }

    companion object{
        private val screenImages = arrayListOf<ImgSrc>()
        init {
            screenImages.add(ImgSrc("blast.jpg",ImgSrc.Slot.s1))
            screenImages.add(ImgSrc("heart.jpg",ImgSrc.Slot.s2))
            screenImages.add(ImgSrc("ship.jpg",ImgSrc.Slot.s3))
            screenImages.add(ImgSrc("watermelon.jpg",ImgSrc.Slot.s4))
            screenImages.add(ImgSrc("orange.jpg",ImgSrc.Slot.s5))
        }
    }

}